### PR TITLE
improve logger types

### PR DESCRIPTION
**Pull request by darky**
## :memo: Description

Now you can use Logger in TypeScript properly.
Also add tsd for typing tests

### :dart: Relevant issues

#453 
#434 

### :gem: Type of change

Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -98,7 +101,10 @@
     "node": ">= 6.x.x"
   },
   "tonicExampleFilename": "docs/runkit/simple.js",
-  "typings": "./index.d.ts",
+  "types": "./index.d.ts",
```
Comment (icebob) [inline package.json]: Is it a common change from `typings` to `types`?

File: package.json [hunk 2]
```diff
@@ -77,6 +79,7 @@
     "redlock": "^3.1.2",
     "thrift": "0.12.0",
     "ts-node": "8.0.2",
+    "tsd": "0.7.0",
```
Comment (icebob) [inline package.json]: `tsd` is a [deprecated library](https://github.com/DefinitelyTyped/tsd).

Comment (darky) [reply package.json]: No, seems you mean this:
https://github.com/DefinitelyTyped/tsd

But used this:
https://github.com/SamVerschueren/tsd

I had seen, that @sindresorhus use it in https://github.com/sindresorhus/type-fest and think, that it good library for testing typescript declaration. Anyway, it better, that before with `hello-world` :)

File: package.json (hunk 1 above)
Comment (darky) [reply package.json]: `tsd` say me to use "types" instead "typings"
Maybe it something related to this:
https://stackoverflow.com/questions/39261204/typings-vs-types-npm-scope

File: package.json (hunk 2 above)
Comment (icebob) [reply package.json]: Ohh, sorry. Yes, I found the other.

File: package.json (hunk 1 above)
Comment (icebob) [reply package.json]: Thanks.

**Pull request merged**